src/async.zsh

```diff
@@ -9,49 +9,59 @@ _zsh_autosuggest_async_request() {
 	typeset -g _ZSH_AUTOSUGGEST_ASYNC_FD _ZSH_AUTOSUGGEST_CHILD_PID
 
 	# If we've got a pending request, cancel it
-	if [[ -n "$_ZSH_AUTOSUGGEST_ASYNC_FD" ]] && { true <&$_ZSH_AUTOSUGGEST_ASYNC_FD } 2>/dev/null; then
-		# Close the file descriptor and remove the handler
-		exec {_ZSH_AUTOSUGGEST_ASYNC_FD}<&-
-		zle -F $_ZSH_AUTOSUGGEST_ASYNC_FD
-
-		# We won't know the pid unless the user has zsh/system module installed
-		if [[ -n "$_ZSH_AUTOSUGGEST_CHILD_PID" ]]; then
-			# Zsh will make a new process group for the child process only if job
-			# control is enabled (MONITOR option)
-			if [[ -o MONITOR ]]; then
-				# Send the signal to the process group to kill any processes that may
-				# have been forked by the suggestion strategy
-				kill -TERM -$_ZSH_AUTOSUGGEST_CHILD_PID 2>/dev/null
+	if (( _ZSH_AUTOSUGGEST_CHILD_PID )); then
+		kill -TERM -- $_ZSH_AUTOSUGGEST_CHILD_PID 2>/dev/null
```

Comment:
I see you've also removed the check of `[[ -o MONITOR ]]`. Was that intentional and if so can you give some reasoning for it? I had never fully tested the different cases there.

Reply:
The check for `MONITOR` is still there. It used to be performed before invoking `kill`, which isn't quite right. I moved it to the proper point (when forking). This makes a difference if the option is changed between forking and killing. I should've this change in the description.

Reply:
Friendly ping.